kuwala/core/canvas/src/components/Blocks/ExportBlock/index.js

```diff
@@ -0,0 +1,66 @@
+import React, {useState} from 'react';
+import {useStoreActions, useStoreState} from "easy-peasy";
+import Block from "../../Common/Block";
+import {triggerExportBlock} from "../../../api/ExportBlock";
+
+const ExportBlock = ({data, id}) => {
+    const {selectedElement} = useStoreState(({ canvas }) => canvas);
+    const { toggleExportConfigModal } = useStoreActions(({ common }) => common);
+    const {
+        setSelectedElementByExportBlockId,
+    } = useStoreActions(({ canvas }) => canvas);
+    const [exportLoading, setExportLoading] = useState(false);
+
+    const openConfigModal = () => {
+        setSelectedElementByExportBlockId(data.exportBlock.exportBlockId)
+        toggleExportConfigModal();
+    }
+
+    // TODO: Create export action by export method
+    const triggerExport = async () => {
+        if (selectedElement) {
+            setExportLoading(true);
+            const res = await triggerExportBlock({
+                exportBlockId: selectedElement.data.exportBlock.exportBlockEntityId
+            });
+
+            let file_name = selectedElement.data.exportBlock.macroParameters.find(el => el.id === 'file_name').value;
+            file_name = file_name.replace(/\.[^/.]+$/, "") // Remove extension
+
+            const url = window.URL.createObjectURL(new Blob([res.data]));
+            const link = document.createElement("a");
+            link.href = url;
+            link.setAttribute(
+                "download",
+                `${file_name}.csv`
+            );
+            document.body.appendChild(link);
+            link.click();
+            link.parentNode.removeChild(link);
+
+            setExportLoading(false);
+        }
+    }
+
+    return (
+        <Block
```

Comment:
Please add the loading property to give the user feedback when the file is being generated.